### PR TITLE
[DROOLS-3609] Add warning message if unsupported DMN + UI improvements 

### PR DESCRIPTION
**Pull request by danielezonca**
DMN unsupported features:
- top level collections
- multiple nested complex objects inside a collection
- multiple nested collections

This file contains all of them [dmn-unsupported.txt](https://github.com/kiegroup/drools-wb/files/2844736/dmn-unsupported.txt)  (please change the extensions to DMN). The warning popup contains the list of the fields unsupported and appears every time user open the scenario.

UI improvements in collection popup:
- Moved "Add Item" outside the scrolling div
- Remove bullet points from ul/li elements
- Improved padding/margin

@kkufova @Rikkola @gitgabrio

### CODE REVIEW

**Comment by gitgabrio** [comment]
jenkins please retest this

**Comment by gitgabrio** [comment]
Fine for me

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationEventHandler.java [hunk 1]
```diff
@@ -359,6 +368,11 @@ public void onEvent(UndoEvent event) {
         }
     }
 
+    @Override
+    public void onEvent(UnsupportedDMNEvent event) {
+        confirmPopupPresenter.show("Unsupported DMN", event.getMessage());
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationEventHandler.java]: Change to:

> Unsupported DMN asset

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java [hunk 2]
```diff
@@ -74,21 +79,51 @@ public boolean isADataType(String value) {
...
+
+        showErrorsAndCleanupState(factModelTuple);
+    }
+
+    private void showErrorsAndCleanupState(FactModelTuple factModelTuple) {
+        StringBuilder builder = new StringBuilder();
+        boolean showError = false;
+        if (factModelTuple.getTopLevelCollectionError().size() > 0) {
+            showError = true;
+            builder.append("Top level collections are not supported:<br/>");
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Change to:

> Top-level collections are not supported! Violated by:

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java [hunk 3]
```diff
@@ -74,21 +79,51 @@ public boolean isADataType(String value) {
...
+            factModelTuple.getTopLevelCollectionError().forEach(error -> builder.append("<b>"+ error + "</b><br/>"));
+            builder.append("<br/>");
+        }
+        if (factModelTuple.getMultipleNestedCollectionError().size() > 0) {
+            showError = true;
+            builder.append("Multiple collections nested are not supported:<br/>");
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Change to:

> Nested collections are not supported! Violated by:

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java [hunk 4]
```diff
@@ -74,21 +79,51 @@ public boolean isADataType(String value) {
...
+            factModelTuple.getMultipleNestedCollectionError().forEach(error -> builder.append("<b>"+ error + "</b><br/>"));
+            builder.append("<br/>");
+        }
+        if (factModelTuple.getMultipleNestedObjectError().size() > 0) {
+            showError = true;
+            builder.append("Multiple nested objects inside a collection are not supported:<br/>");
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Change to:

> Nested objects inside a collection are not supported! Violated by:

**Comment by danielezonca** [comment]
jenkins retest this please

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationEventHandler.java (hunk 1 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationEventHandler.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java (hunk 2 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java (hunk 3 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java (hunk 4 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/DMNDataManagementStrategy.java]: Done

**Comment by danielezonca** [comment]
@manstis 

Can you please squash and merge?

**Pull request merged**